lib/replicator.js

```diff
@@ -2275,12 +2295,15 @@ module.exports = class Replicator {
     protomux.stream.setMaxListeners(0)
     protomux.stream.on('close', this._onstreamclose)
 
+    if (useSession) this._peerSessions++
```

Comment:
Detail: what do you think of adding this line just before `ensureSession`? I think those 2 statements are related, so that makes it clearer (I think it makes no sense to call `this.ensureSession` without incrementing `this._peerSessions`, so we could also put it before `this.ensureSession()`. That way there's an invariant: 'ensureSession is only ever called if this._peerSessions > 0')

Reply:
no its related to the line below, ie, its to avoid the core closing underneath we dbl increment here.